pkg/reconciler/instances/istio/action.go

```diff
@@ -41,5 +42,31 @@ func (a *ReconcileAction) Run(version, profile string, config map[string]interfa
 		return errors.Wrap(err, "Could not patch MutatingWebhookConfiguration")
 	}
 
+	_, err = context.KubeClient.Deploy(context.Context, generateNewManifestWithoutIstioOperatorFrom(manifest.Manifest), istioNamespace, nil)
+	if err != nil {
+		return errors.Wrap(err, "Could not deploy Istio resources")
+	}
+
 	return nil
 }
+
+func generateNewManifestWithoutIstioOperatorFrom(manifest string) string {
+	separator := "---"
```

Comment:
It smells. Can't you parse it (or use k8s client to do it) and remove IstioOperator resource from the list?

Reply:
You can use our K8s-API features to convert manifests into `[]unstructured.Unstructured` entities and operate on them... see https://github.com/kyma-incubator/reconciler/blob/main/pkg/reconciler/kubernetes/kubeclient/unstructured.go#L15